OpenRA.Mods.Common/Traits/Render/WithWallSpriteBody.cs

```diff
@@ -97,8 +104,8 @@ public void Tick(Actor self)
 			adjacent = 0;
 			foreach (var a in adjacentActors)
 			{
-				var rb = a.TraitOrDefault<WithWallSpriteBody>();
-				if (rb == null || rb.wallInfo.Type != wallInfo.Type)
+				var rb = a.TraitOrDefault<IWallConnector>();
+				if (rb == null || !rb.AdjacentWallCanConnect(self, self.Location, wallInfo.Type))
```

Comment:
This first arg needs to be `a` (since you're talking to a trait on the gate actor, not the wall).